### ISSUE TITLE
Custom Vars not updated after upgrade

### ISSUE THREAD

**Issue by Crunsher** [Icinga/icinga2#6569]
When upgrading from Icinga 2.8.4 to 2.9.X (unconfirmed but suspected for other upgrade paths), users observed Hosts not having their custom var changes written to the DB. See [this post](https://monitoring-portal.org/t/icinga-web-2-2-6-0-angepasste-variablen-not-showing-in-host-tab/3549/21)

Also:
* So far I was unable to reproduce this
* Only Hosts seem to be affected at the moment
* Only Hosts which were created before the upgrade have this problem

**Comment by Crunsher** [comment]
Here are sql dumps of a very minimal config from 2.8.4 and 2.9.1: [sqldump.tar.gz](https://github.com/Icinga/icinga2/files/2315067/sqldump.tar.gz)

While no config has been changed, the hosts have different config checksums, this may have happened because of low level changes to our arrays and maps can change the order of elements.

**Comment by Crunsher** [comment]
This seems unrelated to modified attributes. 
1. Changed attributes over API
2. Upgraded to 2.9.1
3. Changed attributes over API -> works

**Comment by Crunsher** [comment]
Possible workaround: Set the config_checksum for Hosts or whatever type is affected to NULL, this will force a complete update of the objects.

```
$ mysql -uroot -picinga icinga
MariaDB [icinga]> UPDATE icinga_hosts SET config_hash = NULL;
MariaDB [icinga]> exit
Bye
$ sudo systemctl restart icinga2
```

Custom vars should now be up to date.

**Comment by dnsmichi** [comment]
We're not able to reproduce this, after days of debugging during workhours. The described workaround with truncating the tables (not only the checksum) after upgrade has been updated, and worked for reporters. Therefore we're stopping with further investigation and closing here.

**Comment by K0nne** [comment]
I found another workaround:

For all affected hosts, you have to add an aditional check to the host objects. after a reload the missing customvar section apperars in icinga web2. after this the dummy-check can be removed.

**Comment by linuxmail** [comment]
hi,

Question: I cloned a host in Icingaweb2 (which was imported via Puppet) and changed some custom host vars. Now I have "original_attributes" with the original cloned customVars, instead of only the new vars. Now I have the problem, that the apply rules doesn't recognize the changes and execute the checks on that host.
host.vars looks OK and shows, what I changed.

I tried both workarounds, without any success:

* Master 2.10.4-1.stretch
* Agent 2.10.4-1.jessie

cu denny
